# CODIGOS PARA HACER LA CONECCION VIA PYTHON A GCP

### Coneccion a Cloud SQL

In [7]:
from google.cloud import storage 
import pandas as pd
import numpy as np
from datetime import datetime
import mysql.connector
import sys

#### Establish Connetion ####

cnx = mysql.connector.connect(user='root', password='soyhenry', host='35.247.37.77', 
                              database='pf_henry')

cursor = cnx.cursor()

#### Connetion Established ####

#### Execute query ####


query1 = ("select * from business")
cursor.execute(query1)


#### Create dataframe from resultant table ####
frame = pd.DataFrame(cursor.fetchall())


frame.head()

0                1             2   3    4    5  6
0  CF33F8-E6oudUQ46HnavjQ   Sonic Drive-In  Ashland City  AZ  2.0    6  1
1  ROeacJQwBeh05Rqg7F6TCg              BAP  Philadelphia  AZ  4.5  205  1
2  bCBPXIVfVzBZBEpFu29dcg  All In Shipping    Belleville  AZ  5.0    5  1
3  JgpnXv_0XhV3SfbfB50nxw      Joe's Pizza         Tampa  AZ  4.0   35  0
4  gbN4RuUiiBxQ2GBFQ2p3Cw    Mr. Margarita       Fernley  AZ  5.0    7  1

#### Viendo el contenido de la tabla Business son 7 columnas y 41444 registros

In [11]:
frame.shape

(41444, 7)

#### Viendo la Informacion de la tabla cargada en el DF

In [8]:
frame.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41444 entries, 0 to 41443
Data columns (total 7 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   0       41444 non-null  object 
 1   1       41444 non-null  object 
 2   2       41444 non-null  object 
 3   3       41444 non-null  object 
 4   4       41444 non-null  float64
 5   5       41444 non-null  int64  
 6   6       41444 non-null  int64  
dtypes: float64(1), int64(2), object(4)
memory usage: 2.2+ MB


#### Haciendo un filtrado para ver solo los negocios de FLORIDA

In [9]:
FL=frame[frame[3]=='FL']
FL

0  \
15115  UJsufbvfyfONHeWdvAHKjA   
15116  jaxMSoInw8Poo3XeMJt8lQ   
15117  0bPLkL0QhhPO5kt1_EXmNQ   
15118  rBmpy_Y1UbBx8ggHlyb7hA   
15119  M0XSSHqrASOnhgbWDJIpQA   
...                       ...   
41439  yqwqLm27A0mxmBx_chTZhg   
41440  Y5gyxnQt44B3axgzrjttlw   
41441  esBGrrmuZzSiECyRBoKvvA   
41442  jsTHopAoQFncD0pGoJQFrw   
41443  wPAaAbFbq5wfLfW0gnYU7g   

                                                       1               2   3  \
15115                                          Marshalls   Land O' Lakes  FL   
15116                                       Adams Dental      Clearwater  FL   
15117                               Zio's Italian Market           Largo  FL   
15118                           Arizona Truck Outfitters          Tucson  FL   
15119                                     Herb Import Co     New Orleans  FL   
...                                                  ...             ...  ..   
41439                                         RadioShack       Havertown  FL   
41440                                 Don Cruz Snowballs          Kenner  FL   
41441                      Colony Grill - St. Petersburg  St. Petersburg  FL   
41442  St John Neumann Center for Rehabilitation & He...    Philadelphia  FL   
41443                                           Intermix    Philadelphia  FL   

         4    5  6  
15115  3.5    6  1  
15116  5.0   10  1  
15117  4.5  100  0  
15118  4.5   10  1  
15119  4.0    5  1  
...    ...  ... ..  
41439  2.5    5  0  
41440  4.5    6  1  
41441  4.5   38  1  
41442  1.5    6  1  
41443  2.0   10  0  

[26329 rows x 7 columns]

### Viendo las Dimenciones despues del Filtrado que son 26329, se ve todo en orden

In [10]:
FL.shape

(26329, 7)

#### Terminando la Conexion

In [12]:
cursor.close()
cnx.close()

### Coneccion a Cloud Storage

In [ ]:
from google.cloud import storage

client = storage.Client()

    # Obtiene el bucket especificado
bucket = client.get_bucket("datasetshenry")

    # Crea un nuevo objeto Blob en el bucket (Nombre de archivo y ruta en CLOUD STORAGE)
blob = bucket.blob("INFO_CONT.parquet")

    # Sube el archivo local al bucket (RUTA ARCHIVO LOCAL)
blob.upload_from_filename("INFO_CONT.parquet")
